dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,551 @@
+
+        Logger.debug(this, ()-> "There's no logged-in user. Processing SAML request...");
+        this.doRequestLoginSecurityLog(request, identityProviderConfiguration);
+
+        final String originalRequest = request.getRequestURI() +
+                (UtilMethods.isSet(request.getQueryString())?
+                    "?" + request.getQueryString() : StringUtils.EMPTY);
+
+        final String redirectAfterLogin = UtilMethods.isSet(request.getParameter(REFERRER_PARAMETER_KEY))
+                ?request.getParameter(REFERRER_PARAMETER_KEY) :
+                // this is safe, just to make a redirection when the user get's logged.
+                originalRequest;
+
+        Logger.debug(this.getClass(),
+                ()-> "Executing SAML Login Redirection with request: " + redirectAfterLogin);
+
+        // if we don't have a redirect yet
+        if (null != session) {
+
+            session.setAttribute(WebKeys.REDIRECT_AFTER_LOGIN, redirectAfterLogin);
+            session.setAttribute(ORIGINAL_REQUEST,             originalRequest);
+        }
+
+        try {
+            // this will redirect the user to the IdP Login Page.
+            DotSamlProxyFactory.getInstance().samlAuthenticationService()
+                    .authentication(request, response, identityProviderConfiguration);
+        } catch (Exception exception) {
+
+            Logger.error(this,  "An error occurred when redirecting to the IdP Login page: " +
+                    exception.getMessage(), exception);
+            Logger.debug(this, ()-> "An error occurred when redirecting to the IdP Login page. Setting 500 " +
+                    "response status.");
+            response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    protected boolean isAnySamlConfigurated() {
+
+        return
+                Try.of(()->this.appsAPI.getAppDescriptor(DotSamlProxyFactory.SAML_APP_CONFIG_KEY,
+                        APILocator.systemUser())).getOrElseGet(e->Optional.empty()).isPresent();
+    }
+
+    protected AutoLoginResult autoLogin(final HttpServletRequest request,
+                                        final HttpServletResponse response,
+                                        HttpSession session,
+                                        final IdentityProviderConfiguration identityProviderConfiguration) {
+
+        final User user          = this.getUser(request, identityProviderConfiguration);
+        boolean continueFilter   = null != user; // by default continue with the filter
+        HttpSession renewSession = session;
+
+        if (continueFilter) {
+            // we are going to do the autologin, so if the session is null,
+            // create it!
+            try {
+
+                Logger.debug(this, "User with ID '" + user.getUserId()
+                        + "' has been returned by SAML Service. User " + "Map: " + user.toMap());
+            } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)